sympy/tensor/array/arrayop.py

```diff
@@ -55,6 +57,14 @@ def tensorproduct(*args):
     if not isinstance(a, NDimArray) or not isinstance(b, NDimArray):
         return a*b
 
+    if isinstance(a, SparseNDimArray) and not isinstance(b, NDimArray):
+        return type(a)({k: b*v for (k, v) in a._sparse_array}, a.shape)
+
+    if isinstance(b, SparseNDimArray) and not isinstance(a, NDimArray):
+        return type(b)({k: a*v for (k, v) in b._sparse_array}, b.shape)
+
+    # TODO: product for two sparse arrays
```

Comment:
don't we only need the product of two sparse arrays?

Reply:
You are right. I will work on the case of two sparse arrays. Thank you for reviewing!